include/ck/tensor_operation/gpu/grid/gridwise_elementwise_1d.hpp

```diff
@@ -0,0 +1,149 @@
+
+namespace ck {
+
+template <typename GridwiseEltwise,
+          typename ADataType,
+          typename BDataType,
+          typename CDataType,
+          typename GridDesc_M0,
+          typename ElementwiseFunctor>
+__global__ void kernel_elementwise_1d(const ADataType* __restrict__ p_a_global,
+                                      const BDataType* __restrict__ p_b_global,
+                                      CDataType* __restrict__ p_c_global,
+                                      const GridDesc_M0 a_grid_desc_m0,
+                                      const GridDesc_M0 b_grid_desc_m0,
+                                      const GridDesc_M0 c_grid_desc_m0,
+                                      const ElementwiseFunctor functor)
+{
+    GridwiseEltwise::Run(p_a_global,
+                         p_b_global,
+                         p_c_global,
+                         a_grid_desc_m0,
+                         b_grid_desc_m0,
+                         c_grid_desc_m0,
+                         functor);
+}
+
+template <typename ADataType,
+          typename BDataType,
+          typename CDataType,
+          typename GridDesc_M0,
+          typename ElementwiseFunctor,
+          index_t ThreadPerBlock,
+          index_t ThreadTileSize,
+          index_t ScalarPerVector>
+struct GridwiseElementwise_1D
+{
+    static constexpr auto I0           = Number<0>{};
+    static constexpr int BlockTileSize = ThreadPerBlock * ThreadTileSize;
+    static constexpr auto thread_desc_M0 =
+        make_naive_tensor_descriptor_packed(make_tuple(Number<ScalarPerVector>{}));
+
+    using PassThrough = tensor_operation::element_wise::PassThrough;
+
+    static __device__ __host__ auto CalculateElementwiseIndex()
+    {
+        const index_t thread_id = get_thread_local_1d_id();
+        const index_t block_id  = get_block_1d_id();
+
+        return make_multi_index(block_id * BlockTileSize + thread_id * ScalarPerVector);
+    }
+
+    __device__ static void Run(const ADataType* __restrict__ p_a_global,
+                               const BDataType* __restrict__ p_b_global,
+                               CDataType* __restrict__ p_c_global,
+                               const GridDesc_M0 a_grid_desc_m0,
+                               const GridDesc_M0 b_grid_desc_m0,
+                               const GridDesc_M0 c_grid_desc_m0,
+                               const ElementwiseFunctor functor)
+    {
+        const auto a_global_buf = make_dynamic_buffer<AddressSpaceEnum::Global>(
+            p_a_global, a_grid_desc_m0.GetElementSpaceSize());
+        const auto b_global_buf = make_dynamic_buffer<AddressSpaceEnum::Global>(
+            p_b_global, b_grid_desc_m0.GetElementSpaceSize());
+        auto c_global_buf = make_dynamic_buffer<AddressSpaceEnum::Global>(
+            p_c_global, c_grid_desc_m0.GetElementSpaceSize());
+
+        StaticBuffer<AddressSpaceEnum::Vgpr, ADataType, ScalarPerVector, true> a_thread_buf;
```

Comment:
Here the concept of `ThreadTileSize` and/or `ScalePerVector` is kind of vague.  Usually, `ThreadSliceSize` is used to indicate the number data to transfer by single `ThreadwiseTensorSliceTransfer` call.   `ScalePerVector` should be different from `ThreadSliceSize` since `ScalePerVector` is required to be on the vector dimension and should completely dividing the length of the vector dimension, but `ThreadSliceSize` can be different. 

Here I suggest: 

1.  Use `ThreadSliceSize` by `ThreadwiseTensorTransfer` and as the number of allocated vgprs
2.  If each thread has more than one slices to process, use some input parameter like `BlockChunkSize` to calculate the num_iter,   like  `num_iter = BlockChunkSize / (ThreadPerBlock * ThreadSliceSize)`.   But you don't have to support more than one slices per thread, since you can always allocate more workgroups.   In your case, `ThreadTileSize` is determined at compile-time,  so it seems you want each thread to handle one Tile/Slice to make the implementation easy

Reply:
This is the thread mapping of implementation of elementwise kerenl
![eltwise1d](https://user-images.githubusercontent.com/9115697/164063062-dfd55205-a582-4341-986a-2f33b5b05c8c.png)

In fact, I borrow the term of threadTile from tensile, which means the number of C element need to be calculated for each thread and this is so called tile size. 
Larger tile size means more works per thread and less work group.
More workgroups also consume the computational resource  
In this implementation, larger the `ThreadTileSize`, larger the iteration here.

`scalrPerVector `is the vector width of buffer_load_dword (buffer_load_dwordx2, buffer_load_dwordx4...etc)
I call `buffer_load_dwordxscalrPerVector`  in each iteration. 
Therefore,  the `SliceLength `and `SrcScalarPerVector `in `ThreadwiseTensorSliceTransfer_v2 `will be the same here.